aws/middleware/user_agent.go

```diff
@@ -74,7 +79,7 @@ type requestUserAgent struct {
 // X-Amz-User-Agent example:
 //
 //	aws-sdk-go-v2/1.2.3 md/GOOS/linux md/GOARCH/amd64 lang/go/1.15
-func newRequestUserAgent() *requestUserAgent {
+func newRequestUserAgent(appID ...string) *requestUserAgent {
```

Comment:
**fix**: `appId` is singular, also this is forcing us to know/care about appId when creating the user agent structure.

There is already a `SDKAgentKeyType` of type `ApplicationIdentifier` for this purpose. We should revert this back and just make an explicit call in setting up the middleware to add two keys. I'll expand elsewhere.